app/cdash/tests/test_starttimefromupload.php

```diff
@@ -0,0 +1,65 @@
+<?php
+require_once dirname(__FILE__) . '/cdash_test_case.php';
+
+use CDash\Model\Build;
+use CDash\Model\Project;
+use Illuminate\Support\Facades\DB;
+
+class StartTimeFromUploadTestCase extends KWWebTestCase
+{
+    private Project $project;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->project = new Project();
+    }
+
+    public function __destruct()
+    {
+        // Delete project & build created by this test.
+        remove_project_builds($this->project->Id);
+        $this->project->Delete();
+    }
+
+    public function testStartTimeFromUpload() : void
+    {
+        $this->deleteLog($this->logfilename);
+
+        // Create test project.
+        $this->login();
+        $this->project->Id = $this->createProject([
+            'Name' => 'StartTimeFromUpload',
+            'NightlyTime' => '18:00:00 America/Denver',
+        ]);
+        $this->project->Fill();
+
+        // Submit our testing data.
+        $file = dirname(__FILE__) . '/data/StartTimeFromUpload/Upload.xml';
+        if (!$this->submission('StartTimeFromUpload', $file)) {
+            $this->fail("Failed to submit {$file}");
+        }
+
+        // No errors in the log.
+        $this->assertTrue($this->checkLog($this->logfilename) !== false);
+
+        // The build exists.
+        $results = DB::select('SELECT id FROM build WHERE projectid = ?', [(int) $this->project->Id]);
```

Comment:
You could use Eloquent for this instead:
```suggestion
        $results = Project::findOrFail((int) $this->project->Id)->builds()->get();
```